src/main/java/software/amazon/awssdk/crt/s3/S3ClientOptions.java

```diff
@@ -113,6 +113,7 @@ public CredentialsProvider getCredentialsProvider() {
     /**
      * The configuration related to signing used by S3 client.
      * `AwsSigningConfig.getDefaultS3SigningConfig(region, credentialsProvider);` can be used as helper to create the default configuration to be used for S3.
+     * If no signing config provided, the client will skip signing.
```

Comment:
Add another comment that signing is required except when  ... (I don't actually know, but you do)